### DISCUSSION TITLE
Redoing release after release publish failed to build [Cake.Recipe]

### DISCUSSION THREAD

**Discussion by MihaMarkic** [cake-build/cake#4290] [Extension Q&A] [answered]
I've tried to publish a release using standard Cake.Recipe procedure (creating draft release and then publish it) but the (appveyor) build failed for some reason.
How would I trigger publish again? Or do I have to increase the version?

**Comment by gep13** [comment, marked as answer]
If you delete the tag from the remote repository, the GitHub release will go back to draft, and then you should be able to Publish the release again.

Do you have a link to the failed build? What was the reason for the failure?

**Reply by MihaMarkic** [reply]
It was on Cake.AppCenter, I think the issue with (my) misconfiguration and hence it didn't build nuget packages.
Can I actually delete the tag (1.4.0) though? It won't allow me and also I can't untag the release.

**Reply by MihaMarkic** [reply]
Ha, nevermind, deleting from CLI is an easy job. If anybody is wondering how to delete a remote tag: `git push --delete origin TAGNAME`

**Reply by gep13** [reply]
Sorry, yes, I should have mentioned. You can't delete the tag through the GitHub UI. Doing it from the command line as you described is what I was referring to.

**Reply by MihaMarkic** [reply]
Can you please take a look at another weird issue with the release - https://ci.appveyor.com/project/cakecontrib/cake-appcenter ?
Not sure where the duplicates come from.
```
========================================
Publish-Release-Packages
========================================
Unable to publish Chocolatey packages. Chocolatey Packages Directory Exists: False
Number of configured Release NuGet Sources: 1
Setting ApiKey in NuGet Push Settings...
Pushing C:/projects/cake-appcenter/BuildArtifacts/Packages/NuGet/Cake.AppCenter.1.4.0.nupkg to Release Source with Url https://api.nuget.org/v3/index.json...
Pushing Cake.AppCenter.1.4.0.nupkg to 'https://www.nuget.org/api/v2/package'...
  PUT https://www.nuget.org/api/v2/package/
  BadRequest https://www.nuget.org/api/v2/package/ 1678ms
The running command stopped because the preference variable "ErrorActionPreference" or common parameter is set to Stop: Response status code does not indicate success: 400 (The package contains one or more duplicated files in the same folder.).
```

**Reply by gep13** [reply]
So, I am not entirely sure how this is happening, or why it is allowed to happen, but looking at the contents of the nupkg file, there is indeed multiple icon.png files:

<img width="605" alt="image" src="https://github.com/cake-build/cake/assets/1271146/4843724a-134b-4cf9-a4af-c02cabb0aa33">

This must be something that nuget.org is checking for, and it is preventing the upload of the file because of it.

**Reply by gep13** [reply]
To resolve this, you will need to remove the explicit reference to the icon.png file in the csproj file, and delete the file from disk:

<img width="3003" alt="image" src="https://github.com/cake-build/cake/assets/1271146/6219f37a-5fac-4b78-ae6b-ab4edc12e71d">

Then, the `CakeContrib.Guidelines` package will do some magic (you would need to confirm with @nils-a about exactly how this works), and put the icon.png file into the package.

That should hopefully allow you to release.

**Reply by nils-a** [reply]
This is strange, "normally" CakeContrib.Guidelines should not add a "second" icon. My first gut feeling is that `PackagePath="\"` throws off my detection logic, but I'll investigate.

As a quick workaround, you have two options:

### use your icon
* Add `<CakeContribGuidelinesIconOmitImport>True</CakeContribGuidelinesIconOmitImport>` to a `<PropertyGroup>`
* This will manually disable the icon "magic" of CakeContrib.Guidelines.

### use the CakeContrib.Guidelines icon
* remove the existing icon
* remove the Include of the icon from the project-file
* remove the PackageIcon from the project-file
* remove the PackageIconUrl from the project-file

**Reply by MihaMarkic** [reply]
@nils-a @gep13 Thanks guys, I went with the second approach and it published.

**Reply by gep13** [reply]
Glad to hear that you got it working! Let me know if you have any further questions!
